src/websocket.js

```diff
@@ -182,11 +168,37 @@ const trades = (payload, cb) => {
         tradeId,
       })
     })
-
-    return w
   })
 
-  return () => cache.forEach(w => w.close())
+  return () => cache.forEach(close => close())
+}
+
+
+const createSocket = (path, cb) => {
+  const onClose = ()=> {
+    if (!closedManually) {
+      w = newSocket(path, cb, onClose)
+    }
+  }
+  
+  let closedManually = false
+  let w = newSocket(path, cb, onClose)  
```

Comment:
`w` should be defined before `onClose` in this scope

Reply:
are you saying it should be *defined* or *declared* before `onClose`? Because it cannot be *defined* before `onClose` as we need to pass `onClose` as one of the parameters to `newSocket()`